app/controllers/pensions/ReliefAtSourcePaymentsAndTaxReliefAmountController.scala

```diff
@@ -0,0 +1,93 @@
+import forms.AmountForm
+import models.mongo.PensionsCYAModel
+import models.pension.reliefs.PaymentsIntoPensionViewModel
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.PensionSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.Clock
+import views.html.pensions.ReliefAtSourcePaymentsAndTaxReliefAmountView
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.Future
+
+
+@Singleton
+class ReliefAtSourcePaymentsAndTaxReliefAmountController @Inject()(implicit val mcc: MessagesControllerComponents,
+                                                                   appConfig: AppConfig,
+                                                                   authAction: AuthorisedAction,
+                                                                   pensionSessionService: PensionSessionService,
+                                                                   errorHandler: ErrorHandler,
+                                                                   view: ReliefAtSourcePaymentsAndTaxReliefAmountView,
+                                                                   clock: Clock) extends FrontendController(mcc) with I18nSupport {
+
+  val amountForm: Form[BigDecimal] = AmountForm.amountForm(
+    emptyFieldKey = "pensions.reliefAtSourceTotalPaymentsAndTaxReliefAmount.error.noEntry",
+    wrongFormatKey = "pensions.reliefAtSourceTotalPaymentsAndTaxReliefAmount.error.invalidFormat",
+    exceedsMaxAmountKey = "pensions.reliefAtSourceTotalPaymentsAndTaxReliefAmount.error.overMaximum"
+  )
+
+  def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit user =>
+    pensionSessionService.getPensionsSessionDataResult(taxYear) {
+      case Some(data) =>
+        if (data.pensions.paymentsIntoPension.rasPensionPaymentQuestion.contains(true)) {
+          data.pensions.paymentsIntoPension.totalRASPaymentsAndTaxRelief match {
+            case Some(amount) =>
+              Future.successful(Ok(view(amountForm.fill(amount), taxYear)))
+            case None => Future.successful(Ok(view(amountForm, taxYear)))
+          }
+        } else {
+          Future.successful(Redirect(ReliefAtSourcePensionsController.show(taxYear)))
+        }
+      case _ =>
+        // TODO: No data navigate to CYA Page to handle when available
+        Future.successful(Redirect(PensionsSummaryController.show(taxYear)))
+    }
+
+  }
+
+  def submit(taxYear: Int): Action[AnyContent] = authAction.async { implicit user =>
+    amountForm.bindFromRequest.fold(
+      formWithErrors => Future.successful(BadRequest(view(formWithErrors, taxYear))),
+      amount => {
+        pensionSessionService.getPensionsSessionDataResult(taxYear) {
+          data =>
+            val pensionsCYAModel: PensionsCYAModel = data.map(_.pensions).getOrElse(PensionsCYAModel(PaymentsIntoPensionViewModel()))
```

Comment:
I'm still a bit unsure about this. But I guess it's fine for now, but this would mean that we are able to add an amount, without the question being answered. I think ti's fine for the question pages, but not sure if it's ok for the amount pages.

Reply:
Yeah, i think that's fine. Happy to approve :)